### ISSUE TITLE
Unusable under FP3 /e/OS 1.21

### ISSUE THREAD

**Issue by schoebel** [organicmaps/organicmaps#7828]
⚠ Have you searched for similar, already existing issues?
yes

**Describe the issue**
Directly after start, a system message box "Organic Maps isn't responding" appears, showing the options "Close app" and "Wait". However, the claim "isn't responding" is wrong, only the reaction time is somewhat slow. By clicking "wait", the message box is always re-appearing after about 4 seconds, permantently in an **endless loop**.

Notice: the app is actually working as expected , but you have to permanetly click on "wait" to be able to do _anything_.

Even worse, switching to any other app does not help, but makes the other app also unusable by permently catching the focus every 4 seconds. So it essentially makes your complete system unusable, als long as you don't abort OrganicMaps.

The previous version 2024.02.06-11-FDroid showed the same message box only once during startup, and never appeared again after clicking on "wait" after the first map was displayed. So the endless loop behaviour is new to version 2024.03.06-4-FDroid.

**Steps to reproduce**
1. Install version 2024.02.06-11-FDroid via FDroid on FairPhone FP3 under /e/OS 1.21-t-20240323388918-stable-FP3 .
2. Install a lot of maps, so that any following fresh startup will need many seconds until the world map (or another map) is displayed, and as a consequence the system message is appearing (as expected).
3. Verify that clicking on "wait" does _not_ lead to the endless loop after the world map (or any zoom) is displayed.
4. Now update to the new version 2024.03.06-4-FDroid and verify that the endless loop is appearing, and that there is no other chance to overcome the problem than killing your app (also via other methods).

**Expected behaviour**
The old behaviour from 2024.02.06-11-FDroid was acceptable.

**System information:**
 - Operating system and its version: e/OS 1.21-t-20240323388918-stable-FP3
 - My default system language is German, and the mentioned messages are in German, but I verified that switching everything to English does not chance the behaviour.
 - Organic Maps version: 2024.02.06-11-FDroid is OK, but 2024.03.06-4-FDroid makes the whole system unusable.
 - Device Model: FairPhone FP3

**Additional context**
Sorry I don't have another system where I could test ist. Probably the issue may be specific to eOS and/or to the hardware.

**Comment by RedAuburn** [comment]
do you have an SD card in your phone? a common issue is that slow SD cards can't load the maps fast enough

**Comment by schoebel** [comment]
Yes, the data is on an SD card, but I would accept the slower speed, and it was fast enough for me in the old version, and it looks OK on the new version.  The "endless loop" behaviour is causing the usability problem...

**Comment by biodranik** [comment]
This behavior is highly likely caused by your SD-card (or SD controller). If you replace it, or store maps in internal memory, the issue should disappear.

**Comment by schoebel** [comment]
Well, I just rebooted the phone. The problem remained present at first.
After several hours just laying around, the problem suddenly disappeard completely. Now everything is fast, and the message boxes have vanished. I didn`t change anything at the SD card level.

Looks like some kind of "reorganization" has ocurred in the background, leading to a kind of "self-fixing" :)

Thanks for your hints.

**Comment by schoebel** [comment]
Nevertheless, I have some new observations.

1) Rebootet once again. Afterwards, the problem re-appeared again (slow performance, the message box, and the endless loop).

I became curious, since I have seen a kind of self-repair, and I want to use OrganicMaps. Thus:

2) Rebooted once again. Afterwards, all problems were gone again. Again, it is fast now, no message boxes, and of course no endless loops.

This means that the problem **is not reproducible**, at least for me, and I have no clue whether it could be seen as a bug / regression, or a hint, or whatever (e.g. some unknown race conditions, or some hidden random number generator, etc)?

For myself, I have a solution for now. Anyone else seeing a similar problem / behaviour?

**Comment by biodranik** [comment]
Some software is actively scanning or writing to your SD card and thus slows down other apps like Organic Maps, which heavily rely on fast reading of the offline maps data.

Or your SD-card/SD-controller are bad.

**Comment by poeticus** [comment]
I can confirm that the issue as described in the original post is reproducible on my Fairphone 3+ running /e/os v.1.21.

The only difference in my current set-up is that I am running OM Version 2024.03.31-8-FDroid.

OM 2024.03.05-4-FDroid ran fine for me under /e/os v.1.20, with just the single 'not responding' message which did not reappear if I pressed 'wait'.

/e/os v1.21 is an upgrade from Android 12 to 13. I performed this upgrade just before updating OM to its latest version.

I hadn't tried OM on the upgraded OS before updating it so cannot say how it would have worked.

At the moment it's a bit hit or miss whether OM will open without the recurring message. When it appears, just selecting 'close' then swiping OM from the recents screen and reopening sometimes works, sometimes an additional clearing of the cache works, sometimes a reboot works, sometimes none of these.

I agree with the earlier comment that the 'not responding' message appears incorrect. If I'm moving the map around when it appears, I can continue to move the map while the message still shows. If OM really wasn't responding I'd expect it to freeze.

**Comment by vng** [comment]
@poeticus @schoebel 
Do ensure, you guys use SD-card or not? Is it possible to _move_ the app and maps into device's internal memory and check?

**Comment by poeticus** [comment]
I use the SD card for map storage. I'm unable to move the maps to internal memory due to insufficient space.

As an ordinary user it doesn't feel to me that this issue is a simple case of accessing the SD card.

If, on app startup, the map loads and reaches the point at which it is displayed without the message appearing then I never see the message subsequently. I can move around the map, zoom in and out to any extent and perform other operations with no problems.

If, however, the message appears before the map is displayed then it is impossible to stop it repeatedly appearing without closing the app. 

The map will load and display in this case if I either repeatedly dismiss the message, or if I leave it on the screen untouched (in which case it appears behind the message box). In either case it then reappears within a few seconds of dismissing it by touching 'wait'.

In the case where the map loads and displays successfully without the message appearing, the only time I do then see it is if I try to download multiple maps, for example for a whole country. I am, however, able to download these one by one with no problem. It's as if the queuing is somehow causing the problem in this case.

**Comment by biodranik** [comment]
99.99% of similar issues from our users were related to bad SD cards.

**Comment by poeticus** [comment]
Mine is a fast card: Sandisk 256GB Extreme Pro microSDXC. 

OM was working fine with it right up until the recent updates.

OM still works fine with it some of the time. 

OsmAnd still works fine with it all of the time.

Just a coincidence that two users have experienced exactly the same issue at the same time here or more likely that there is another underlying issue?

**Comment by poeticus** [comment]
Another related thing I've found which feels strange relates to the case where the message box has appeared but I've got to the stage where the map is displayed.

If I close OM by touching the 'close' button it closes fine and the message stops appearing.

If, however, I touch 'wait', then close OM by swiping left and then remove it from the 'recents' screen the message continues to reappear. Only touching 'close' in the message box will stop this.

**Comment by vng** [comment]
Hm, could you please check "Enable logging" option in the app? Is it on or off?

If ON, please send us the logs (via "Report the problem") and turn it OFF to observe the behavior.

**Comment by schoebel** [comment]
> I use the SD card for map storage. I'm unable to move the maps to internal memory due to insufficient space.

Exactly the same problem at my side.

> > As an ordinary user it doesn't feel to me that this issue is a simple case of accessing the SD card.

Same from my side (as I tried to explain). I saw _multiple_ cases where it suddenly was fast without problems, but after re-start of the app it was slow again and the problem re-appeared. I cannot see an explanation based on SD card IO performance & co.

> If, on app startup, the map loads and reaches the point at which it is displayed without the message appearing then I never see the message subsequently. I can move around the map, zoom in and out to any extent and perform other operations with no problems.
> 
> If, however, the message appears before the map is displayed then it is impossible to stop it repeatedly appearing without closing the app.

This is exactly what I described, and thanks to poeticus to take the time and the effort to veryfiy this. Hopefully the problem will be treated by the devs as a "bug" in some sense (from a user's viepoint), due to practical unusablity (eating a lot of time and effort to circumvent the problem in a "random" manner).

**Comment by biodranik** [comment]
We need more information: does the issue disappear when maps are stored in internal memory, or not?

**Comment by poeticus** [comment]
@vng I understand the reason for this, but from a privacy perspective I'm not really comfortable submitting logs without properly understanding what information they will contain. I have 'enable logging' set to off.

Interested to find this (apparently resolved) issue from July 2023 which details what appears to be exactly the same issue on a Fairphone 3 after updating to Android 13 - a repeated 'not responding' message every few seconds but OM appearing to fully work regardless.
https://github.com/organicmaps/organicmaps/issues/5596

The recent update to /e/os v1.21-t on the Fairphone 3 was the /e/os upgrade from Android 12 to 13. Is it possible there could be a link here?

**Comment by vng** [comment]
Ok, no problem. Please, keep logging OFF and watch how it works. From our perspective, we do not share logs anywhere, and the only sensitive info there are possible GPS coordinates.

**Comment by poeticus** [comment]
Updated feedback after 2-3 days further testing and now I've found what appears to be a reliable workaround to get OM starting & working OK, and to summarise how OM is currently working here in general.

1. When the "not responding" message appears on starting OM I've now found that pressing "close" on the message's first appearance and then immediately restarting from the app icon (without first removing OM from the 'recents' screen) results in it then starting and opening the map OK most times.

2. If this doesn't work first time, repeating the closing & restart usually does. On the occasions it doesn't, it seems to work by around the 4th attempt.

3. I now find this to be the easiest and most consistently successful approach, with no need to clear cache, restart the phone etc. As noted previously, once OM has opened the map with no message having appeared it continues to work fine.

4. After the phone's not been used for 6-7 hours overnight OM starts OK first time with no "not responding" message. This is as noted in the original posts.

5. During the day as the phone is being used although OM does sometimes start & work OK first time without the message, most times it doesn't.

**Comment by biodranik** [comment]
Do you observe the noted behavior when maps are stored in the internal device memory?

**Comment by poeticus** [comment]
> I use the SD card for map storage. I'm unable to move the maps to internal memory due to insufficient space.

**Comment by brezelman** [comment]
I'm using (shared) internal storage for maps and the same behavior occurred on my FairPhone 4 with Android 13.

**Comment by biodranik** [comment]
Thanks @brezelman , can you please test if the previous version works fine? Please don't forget to export/backup all your bookmarks and tracks before uninstalling Organic Maps.

**Comment by brezelman** [comment]
Ok, I went back to Version 2024.03.05-4-FDroid and redownloaded the maps I had before. Trying to reproduce the problem right now. No "luck" so far.

**Comment by brezelman** [comment]
I could get it to show the "not responding" popup once today, but couldn't reproduce it afterwards.

**Comment by brezelman** [comment]
Confirmed reproducible now with recurring popups on Version 2024.03.05-4-FDroid. Will try current rc next.

**Comment by poeticus** [comment]
I've now updated to OM 2024.05.11-3-FDroid and also to /e/os v2.0t and the issue still persists.

I have, however, found a workaround that at the moment consistently enables the map to open and be used without the recurring 'not responding' message appearing. I've not yet had this fail.

1. When opening OM, once the green splash screen appears swipe up and put OM into the background. 

2. After a few seconds the 'not responding' message briefly appears but then goes. 

3. If I then wait for the time I know it would take for the maps to load and then reopen OM, the map appears (albeit at full World view not at last view) and the 'not responding' message does not reappear. I can then use the map fully with no issues.

It appears that whatever is causing the message to repeatedly reappear is killed by having OM in the background rather than open in the foreground when it is first triggered.

**Comment by biodranik** [comment]
Can you please record full system logcat logs via the cable? If no, then please enable logs in OM settings, and kill the app.

Enable screen recording, launch OM, reproduce the issue, and share logs with us at android at organicmaps app or via Report a bug button, and attach screen video and the log so we can see the time moments when it happens.

**Comment by vng** [comment]
@schoebel @poeticus @brezelman Do you have the same issue with the latest update?

**Comment by brezelman** [comment]
> [@schoebel](https://github.com/schoebel) [@poeticus](https://github.com/poeticus) [@brezelman](https://github.com/brezelman) Do you have the same issue with the latest update?

I could reproduce it twice but without logcat enabled.

**Comment by brezelman** [comment]
Opened the app and tried to move the map view, but GPS was disabled, so I had to close the popup telling me about it. But even before that the crash occurred again.

> ANR in app.organicmaps (app.organicmaps/.DownloadResourcesActivity)
PID: 13325
Reason: Input dispatching timed out (568c216 app.organicmaps/app.organicmaps.DownloadResourcesActivity (server) is not responding. Waited 5003ms for FocusEvent(hasFocus=true)).
Parent: app.organicmaps/.DownloadResourcesActivity
ErrorId: 368ff4a5-24ee-4824-a2c9-0ad07456fae9
Frozen: false

**Comment by biodranik** [comment]
@brezelman Did you try the latest FDroid build? Please try with our latest web build (it can be installed separately): https://github.com/organicmaps/organicmaps/releases/tag/2025.10.23-22-android

**Comment by brezelman** [comment]
> Did you try the latest FDroid build?

Yes, this was with Version 2025.10.23-22-FDroid.

**Comment by biodranik** [comment]
@brezelman thanks for confirming! Could you try the web apk to check one hypothesis about the bug? https://github.com/organicmaps/organicmaps/releases/tag/2025.10.23-22-android

**Comment by brezelman** [comment]
Oh, I forgot to select intern device as map storage location after reinstallation. Will try to reproduce it with internally saved map.

**Comment by biodranik** [comment]
Can you please try the latest version? We made some changes there, maybe they can help. 

So far the issue reproduces _only_ if downloaded maps are stored on the SD card, and _does not_ reproduce if maps are downloaded to the internal memory, correct?

**Comment by brezelman** [comment]
> So far the issue reproduces _only_ if downloaded maps are stored on the SD card, and _does not_ reproduce if maps are downloaded to the internal memory, correct?

No, I could reproduce it again now with maps on internal storage on 2025.10.13 release. I sent log by e-mail again.

Will try with new version.

**Comment by biodranik** [comment]
Thanks for the logs, It looks like the culprit is a slow SD card. Removing it completely and trying again should likely help (can you please confirm that?).

The ANR (caused by the blocked main thread) arises not only when SD card is accessed, but also when OM _scans_ for available storages, even before it starts using them. That may be tricky to fix, as the map storage should be set up before everything else is done (we'll see what can be done).

As it is frequently reproduced on FP3, it looks like a system-specific issue (hardware or software).

```
11-09 23:41:38.668 (main) I/StoragePathManager: StoragePathManager.java:235 addStorageOption(): Accepted /storage/emulated/0/Android/data/app.organicmaps.web/files/ - currently configured, external, 24605446144 available of 110382735360 bytes, emulated, state=mounted, primary, label='Interner gemeinsamer Speicher'
11-09 23:41:38.680 (main) D/StorageUtils: StorageUtils.java:30 isDirWritable(): Checking for writability /storage/BF36-1709/Android/data/app.organicmaps.web/files
11-09 23:41:41.608 (main) I/StoragePathManager: StoragePathManager.java:235 addStorageOption(): Accepted /storage/BF36-1709/Android/data/app.organicmaps.web/files/ - external, 36161716224 available of 62243930112 bytes, removable, state=mounted, uuid=BF36-1709, label='SD-Karte'
11-09 23:41:41.609 (main) D/StorageUtils: StorageUtils.java:30 isDirWritable(): Checking for writability /data/data/app.organicmaps.web/files
```

**Comment by brezelman** [comment]
> That may be tricky to fix, as the map storage should be set up before everything else is done (we'll see what can be done).

Thanks for your effort, even though just we Fairphone users are affected.
The check has to be executed on startup and on the main thread even though the user selected the local storage as map dir?

**Comment by biodranik** [comment]
At the moment, yes.
